scripts/ci/run_perftest.sh

```diff
@@ -68,27 +78,81 @@ go get \
+systemctl start influxdb
+
 # Run and record tests
 datestring=$(date +%s)
 seed=$datestring
+db_name="benchmark_db"
 for scale in 50 100 500; do
   # generate bulk data
   scale_string="scalevar-$scale"
   scale_seed_string="$scale_string-seed-$seed"
   data_fname="influx-bulk-records-usecase-devops-$scale_seed_string.txt"
   $GOPATH/bin/bulk_data_gen --seed=$seed --use-case=devops --scale-var=$scale --format=influx-bulk > ${DATASET_DIR}/$data_fname
+  query_files=""
+  for format in http flux-http; do
+    query_fname="query-devops-$scale_seed_string-$format.txt"
+    $GOPATH/bin/bulk_query_gen --seed=$seed --use-case=devops --scale-var=$scale --format=influx-$format --db $db_name --queries 10000 --query-type 8-host-1-hr > ${DATASET_DIR}/$query_fname
+    query_files="$query_files $query_fname"
+  done
 
-  # run ingest tests
-  test_type=ingest
+  # Test loop
   for parseme in "5000:2" "5000:20" "15000:2" "15000:20"; do
     batch=$(echo $parseme | cut -d: -f1)
     workers=$(echo $parseme | cut -d: -f2)
+
+    # generate load data
     load_opts="-batch-size=$batch -workers=$workers -urls=http://${NGINX_HOST}:8086 -do-abort-on-exist=false -do-db-create=true -backoff=1s -backoff-timeout=300m0s"
     if [ -z $INFLUXDB2 ] || [ $INFLUXDB2 = true ]; then
       load_opts="$load_opts -organization=$TEST_ORG -token=$TEST_TOKEN"
     fi
 
-    cat ${DATASET_DIR}/$data_fname | $GOPATH/bin/bulk_load_influx $load_opts | jq ". += {branch: \"${INFLUXDB_VERSION}\", commit: \"${TEST_COMMIT}\", time: \"$datestring\", i_type: \"${DATA_I_TYPE}\"}" > $working_dir/test-$test_type-$scale_string-batchsize-$batch-workers-$workers.json
+    # run ingest tests
+    cat ${DATASET_DIR}/$data_fname | $GOPATH/bin/bulk_load_influx $load_opts | jq ". += {branch: \"${INFLUXDB_VERSION}\", commit: \"${TEST_COMMIT}\", time: \"$datestring\", i_type: \"${DATA_I_TYPE}\"}" > $working_dir/test-ingest-$scale_string-batchsize-$batch-workers-$workers.json
+
+    bucket_id=$(influx bucket list --host http://${NGINX_HOST}:8086 -n ${db_name} -o $TEST_ORG --token $TEST_TOKEN | head -2 | tail -1 | cut -d '	' -f1)
+    sleep 1
+    # stop daemon and force compaction
+    systemctl stop influxdb
+    set +e
+    shards=$(find /var/lib/influxdb/engine/data/$bucket_id/autogen/ -maxdepth 1 -mindepth 1)
+    set -e
+    for shard in $shards; do
+      /home/ubuntu/influx_tools compact-shard -force -verbose -path $shard
+    done
+
+    # restart daemon and wait til it's online
+    systemctl unmask influxdb.service
+    systemctl start influxdb
+    i=0
+    set +e
+    HTTP_CODE=$(curl -s --output $(mktemp) --write-out "%{http_code}" -I http://localhost:8086/ping)
+    while [ $HTTP_CODE != "204" ]; do
+      if [ $i -ge 30 ]; then
+        echo "influxd not responding to pings after 30 seconds"
+        exit 1
+      fi
+      i=$((i+1))
+      sleep 1
+      HTTP_CODE=$(curl -s --output $(mktemp) --write-out "%{http_code}" -I http://localhost:8086/ping)
+    done
+    set -e
```

Comment:
Does #21864 remove the need for this ping-loop?

Reply:
yes, I'll update